Objective-C/Internal/CBLFLDict.h

```diff
@@ -0,0 +1,30 @@
+//
+//  CBLFLDict.h
+//  CouchbaseLite
+//
+//  Created by Pasin Suriyentrakorn on 4/23/17.
+//  Copyright © 2017 Couchbase. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "Fleece+CoreFoundation.h"
+@class CBLC4Document;
+@class CBLDatabase;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface CBLFLDict : NSObject
```

Comment:
I, for one, will not care if I have to port to C# from C++ or Objective-C.  C++ might even be easier because Objective-C collections have always annoyed me because they are non-generic.  Just be sure not to make use of C++ specific things like multiple inheritance from non-interface classes though (doubt that will be an issue here).